src/main/java/ch/njol/skript/expressions/ExprExperience.java

```diff
@@ -95,42 +98,45 @@ public Class<?>[] acceptChange(final ChangeMode mode) {
 	
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
-		double d;
-		if (e instanceof ExperienceSpawnEvent)
-			d = ((ExperienceSpawnEvent) e).getSpawnedXP();
-		else if (e instanceof BlockBreakEvent)
-			d = ((BlockBreakEvent) e).getExpToDrop();
-		else
+		double eventExp;
+		if (e instanceof ExperienceSpawnEvent) {
+			eventExp = ((ExperienceSpawnEvent) e).getSpawnedXP();
+		} else if (e instanceof BlockBreakEvent) {
+			eventExp = ((BlockBreakEvent) e).getExpToDrop();
+		} else if (e instanceof PlayerExpChangeEvent) {
+			eventExp = ((PlayerExpChangeEvent) e).getAmount();
+		} else {
 			return;
-		
-		if (delta != null)
-			for (final Object o : delta) {
-				final double v = o instanceof Experience ? ((Experience) o).getXP() : ((Number) o).doubleValue();
-				switch (mode) {
-					case ADD:
-						d += v;
-						break;
-					case SET:
-						d = v;
-						break;
-					case REMOVE:
-					case REMOVE_ALL:
-						d -= v;
-						break;
-					case RESET:
-					case DELETE:
-						assert false;
-						break;
-				}
+		}
+		for (Object obj : delta) {
+			double value = obj instanceof Experience ? ((Experience) obj).getXP() : ((Number) obj).doubleValue();
+			switch (mode) {
+				case ADD:
+					eventExp += value;
+					break;
+				case SET:
+					eventExp = value;
+					break;
+				case REMOVE:
+				case REMOVE_ALL:
```

Comment:
Basically `REMOVE_ALL` shouldn't be support but since it isn't from your PR, I will just ignore this to avoid Breaking Changes but I think it should be supported